lib/mongoid/config.rb

```diff
@@ -125,6 +125,13 @@ module Config
     # always return a Hash.
     option :legacy_attributes, default: false
 
+    # When this flag is true, Mongoid will validate the type of the value
+    # assigned to a field, such that it can be coerced to a value of the field
+    # type. If it is "uncastable," an InvalidValue error will be raised. If this
+    # feature flag is off, nil will be written to the attribute and the database
+    # on save.
+    option :validate_attribute_types, default: true
```

Comment:
> For demongoize (reading from DB), the behavior needs to match the Mongoid 7.x behavior, we should not raise errors. 

There is a use case for failing demongoization. It is described in https://jira.mongodb.org/browse/MONGOID-5382.

> Otherwise, entire queries may fail due to just one bad field in one document.

This is something I will verify as part of the ongoing design work for https://jira.mongodb.org/browse/MONGOID-5332 but my impression is this statement is not correct - demongoization is only performed on field access and thus failing demongoization does not result in failing queries since demongoization happens after the query is already completed. If Mongoid does not already behave in this manner we are considering how to obtain this behavior, by default or via opt-in.